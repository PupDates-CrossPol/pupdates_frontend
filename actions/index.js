export const setUserInfo = (user) => ({
  type: 'SET_USER_INFO',
  user,
})

export const setPackInfo = pack => ({
  type: 'SET_PACK_INFO',
  pack
})

export const setPackPhotos = packPhotos => ({
  type: 'SET_PACK_PHOTOS',
  packPhotos
})

export const setOtherUsers = otherUsers => ({
  type: 'SET_OTHER_USERS',
  otherUsers
})

export const setSwipeUser = swipeUser => ({
  type: 'SET_SWIPE_USER',
  swipeUser
})

export const setSwipePack = swipePack => ({
  type: 'SET_SWIPE_PACK',
  swipePack
})

export const setSwipePackPhotos = swipePhotos => ({
  type: 'SET_SWIPE_PACK_PHOTOS',
  swipePhotos
})

export const setTempUserImage = tempUserImage => ({
	type: 'SET_TEMP_USER_IMAGE',
	tempUserImage
})

export const setImageUpload = imageUpload => ({
	type: 'SET_IMAGE_UPLOAD',
	imageUpload
})

export const setModalState = modalState => ({
	type: 'SET_MODAL_STATE',
	modalState
})

export const setMatches = matches => ({
	type: 'SET_MATCHES',
	matches
})

export const setMatchesPack = matchesPack => ({
	type: 'SET_MATCHES_PACK',
	matchesPack
})

export const setMatchesPackImages = matchesImages => ({
	type: 'SET_MATCHES_PACK_IMAGES',
	matchesImages
})