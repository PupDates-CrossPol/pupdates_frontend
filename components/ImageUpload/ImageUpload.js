import * as React from 'react';
import firebase from 'firebase';
import * as ImagePicker from 'expo-image-picker';
import * as Permissions from 'expo-permissions';
import * as Camera from 'expo-camera';
import { LinearGradient } from 'expo-linear-gradient';
import { StyleSheet, Text, View, Image, TextInput, Button, TouchableOpacity, ScrollView, SafeAreaView } from 'react-native';
import * as apiCalls from '../../apiCalls';
import { connect } from 'react-redux';
import { setUserInfo, setPackInfo, setPackPhotos, setTempUserImage, setImageUpload } from '../../actions';
import ApiKeys from '../../ApiKeys';

firebase.initializeApp(ApiKeys.firebaseConfig);

class ImageUpload extends React.Component {
	state = {
    id: '',
    images: [],
    loading: false
  }

  render() {
  	return (
  		<SafeAreaView>
      {this.state.loading && <Text>Loading</Text>}
        <TouchableOpacity style={styles.button} onPress={() => this.selectImg()}>
          <LinearGradient
            colors={['orange', '#c32525']}
            style={styles.linearGradient}
          >
              <Text style={styles.buttonText}>Choose From Library</Text>
          </LinearGradient>
        </TouchableOpacity>
        <TouchableOpacity style={styles.button} onPress={() => this.takeImg()}>
          <LinearGradient
            colors={['orange', '#c32525']}
            style={styles.linearGradient}
          >
              <Text style={styles.buttonText}>Take A Photo</Text>
          </LinearGradient>
        </TouchableOpacity>
        <TouchableOpacity style={styles.button} onPress={() => this.addImg()}>
          <LinearGradient
            colors={['orange', '#c32525']}
            style={styles.linearGradient}
          >
              <Text style={styles.buttonText}>Submit</Text>
          </LinearGradient>
        </TouchableOpacity>
        <TouchableOpacity style={styles.button} onPress={() => this.props.setImageUpload(null)}>
          <LinearGradient
            colors={['gray', 'black']}
            style={styles.linearGradient}
          >
              <Text style={styles.buttonText}>Cancel</Text>
          </LinearGradient>
        </TouchableOpacity>
  		</SafeAreaView>
  	)
  }

setTimeout

  selectImg() {
    this.setState({ id: '' + Math.random().toString(36).substr(2, 9) });
    ImagePicker.requestCameraRollPermissionsAsync()
      .then(response => {
        if (response.granted === true) {
          ImagePicker.launchImageLibraryAsync()
            .then(response => {
            	this.uploadImg(response.uri, `${this.state.id}`);
            	this.props.setTempUserImage(response.uri);
              setTimeout(this.addImg, 10000)
            }
        )
      }
    })
      .catch(error => console.log(error))
  }

  takeImg() {
    this.setState({ id: '' + Math.random().toString(36).substr(2, 9) });
    Camera.requestPermissionsAsync()
      .then(response => {
        if (response.granted === true) {
          ImagePicker.launchCameraAsync()
            .then(response => {
            	this.uploadImg(response.uri, `${this.state.id}`);
            	this.props.setTempUserImage(response.uri);
              setTimeout(this.addImg, 10000)
            }
        )
      }
    })
      .catch(error => console.log(error))
  }

  uploadImg = async (uri, imageId) => {
    this.setState({loading: true})
    const response = await fetch(uri);
    const blob = await response.blob();

    let ref = firebase.storage().ref().child(`images/${imageId}`);
    return ref.put(blob)
  }

  addImg = async () => {
    const { id, email, first_name, last_name, description } = this.props.user
    this.props.setTempUserImage(null);
    const url = await firebase.storage().ref().child(`images/${this.state.id}`).getDownloadURL();
    this.setState({ images: [...this.state.images, url]});
    this.props.setUserInfo({ description, email, first_name, id, last_name, photo: url })
    const user = await apiCalls.patchUserPhoto(url, id)
    this.props.setImageUpload(null);
    this.setState({loading: false})
  }
}

const styles = StyleSheet.create({
  linearGradient: {
    width: '80%',
    height: 40,
    marginTop: 5,
    alignItems: 'center',
    justifyContent: 'center',
  },
  button: {
    // borderRadius: 50,
    borderTopRightRadius: 20,
    borderBottomRightRadius: 20,
    borderBottomLeftRadius: 20,
    borderTopLeftRadius: 20,
    alignItems: 'center',
    justifyContent: 'center',
    color: 'black'
  },
  buttonText: {
    color: '#fff',
    fontSize: 25,
  }
})

export const mapStateToProps = state => ({
  user: state.user,
  pack: state.pack,
  packPhotos: state.packPhotos,
  tempUserImage: state.tempUserImage,
  imageUpload: state.imageUpload
})

export const mapDispatchToProps = dispatch => ({
  setUserInfo: (userInfo) => dispatch(setUserInfo(userInfo)),
  setPackInfo: (dogPack) => dispatch(setPackInfo(dogPack)),
  setPackPhotos: (dogPackPictures) => dispatch(setPackPhotos(dogPackPictures)),
  setTempUserImage: (tempUserImage) => dispatch(setTempUserImage(tempUserImage)),
  setImageUpload: (imageUpload) => dispatch(setImageUpload(imageUpload))
})

export default connect (mapStateToProps, mapDispatchToProps)(ImageUpload)