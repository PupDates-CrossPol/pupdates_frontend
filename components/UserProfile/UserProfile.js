import * as React from 'react';
import * as apiCalls from '../../apiCalls';
import { StyleSheet, Text, View, Image, TextInput, Button, TouchableOpacity, ScrollView, SafeAreaView, Modal } from 'react-native';
import { Ionicons } from '@expo/vector-icons';
import { connect } from 'react-redux';
import { setUserInfo, setPackInfo, setPackPhotos, setTempUserImage, setImageUpload, setModalState } from '../../actions';
import ImageUpload from '../ImageUpload/ImageUpload';

export class UserProfileScreen extends React.Component {
    static navigationOptions = ({navigation}) => ({        
        headerLeft: () => (
          <TouchableOpacity style={styles.backArrow} onPress={() => navigation.navigate('Menu')}>
            <Ionicons name="ios-arrow-back" size={30} color='rgb(53, 129, 252)' />
          </TouchableOpacity>
          ),
        headerTitle: () => <Image source={require('../../assets/ProfileSpread.png')} style={styles.navTitle}/>,
        headerRight: () => <Image source={require('../../assets/PupDatesLogo.png')} style={styles.logo} />,
        headerStyle: {
          backgroundColor: '#fff',
        },
        headerTintColor: '#fff',
        headerTitleStyle: {
          fontWeight: 'bold',
        },
        headerLeftContainerStyle: styles.leftNavIcon,
        headerRightContainerStyle: styles.rightNavIcon,
      })

    render() {
      const modalBackgroundStyle = {
            backgroundColor: 'rgba(0, 0, 0, 0.2)'
          };
          console.log(this.props.user)
        return (
            <SafeAreaView>
              <ScrollView>
                <View style={styles.componentTitleHeader}>
                    <TouchableOpacity style={styles.imageBtn} onPress={() => 
                      this.props.setModalState(this.props.modalState)
                    }>
                      {this.props.tempUserImage && <Image source={{uri: this.props.tempUserImage}} style={styles.userImage}/>}
                      {!this.props.tempUserImage && <Image source={{uri: this.props.user.image}} style={styles.userImage}/>}
                    </TouchableOpacity>
                    <Text style={styles.infoHeader}>About Me:</Text>
                    <Text style={styles.userInfoText}>First Name: {this.props.user.first_name}</Text>
                    <Text style={styles.userInfoText}>Last Name: {this.props.user.last_name}</Text>
                    <Text style={styles.userInfoText}>Email: {this.props.user.email}</Text>
                    <Text style={styles.infoHeader}>Description:</Text>
                    <Text style={styles.userInfoText}>{this.props.user.description}</Text>
                    <Modal
                          animationType="slide"
                          transparent={true}
                          visible={this.props.modalState}
                          >
                      <View style={[styles.modalContainer, modalBackgroundStyle]}>
                          <ImageUpload currentComponent = {'User'}/>
                      </View>
                  </Modal>
                </View>
              </ScrollView>
            </SafeAreaView>
        )
    }
}

const styles = StyleSheet.create({
    container: {
      flex: 1,
      backgroundColor: '#fff',
      alignItems: 'center',
      marginTop: 90
    },
    logo: {
        height: 40,
        width: 40,
    }, 
    image: {
      height: 140,
      width: 140,
    },
    title: {
      fontSize: 50,
    },
    backArrow: {
      marginLeft: 5,
    },
    imageBtn: {
      height: 265,
      alignItems: 'center',
      justifyContent: 'center',
      borderBottomColor: 'rgba(0,0,0,0.57)',
      borderBottomWidth: 1,
      shadowColor: "#000", 
      shadowOffset: { width: 0, height: 2, }, 
      shadowOpacity: 0.25, 
      shadowRadius: 3.84, 
      elevation: 5,
      marginLeft: 55,
      marginRight: 55
    },
    userImage: {
      aspectRatio: 1/1,
      height: 199,
      borderRadius: 100,
      borderColor: 'black',
      borderWidth: 2,
    },
    userInfoText: {
      fontSize: 20,
      fontWeight: '300',
      marginTop: 5,
      paddingLeft: 65,
      color: 'rgba(0,0,0,0.57)',
    },
    infoHeader: {
      fontSize: 25,
      color: 'rgb(21, 112, 125)',
      marginLeft: 5,
      paddingLeft: 15,
      marginBottom: 20,
      marginTop: 20,
    },
    modalContainer: {
        flex: 1,
        alignItems: 'center',
        justifyContent: 'center',
        backgroundColor: '#ecf0f1',
    },
    input: {
      height: 30,
      width: '70%',
      borderColor: 'lightgrey',
      borderRadius: 50,
      borderWidth: 1.5,
      padding: 10,
      height: 60,
      margin: 20,
      alignItems: 'center'
    },
    linearGradient: {
      width: '20%',
      height: 50,
      alignItems: 'center',
      justifyContent: 'center',
    },
    button: {
      borderRadius: 100,
      alignItems: 'center',
      justifyContent: 'center',
      color: 'black',
      marginTop: 0
    },
    buttonText: {
      color: '#fff',
      fontSize: 15,
    },
    navTitle: {
        width: 160,
        height: 40,
        marginBottom: 5,
    },
    rightNavIcon: { 
        marginRight: 20,  
        marginBottom: 10, 
    },
    leftNavIcon: { marginLeft: 10,  marginBottom: 5, }
  });

export const mapStateToProps = state => ({
  user: state.user,
  pack: state.pack,
  packPhotos: state.packPhotos,
  tempUserImage: state.tempUserImage,
  imageUpload: state.imageUpload,
  modalState: state.modalState
})

export const mapDispatchToProps = dispatch => ({
  setUserInfo: (userInfo) => dispatch(setUserInfo(userInfo)),
  setPackInfo: (dogPack) => dispatch(setPackInfo(dogPack)),
  setPackPhotos: (dogPackPictures) => dispatch(setPackPhotos(dogPackPictures)),
  setTempUserImage: (tempUserImage) => dispatch(setTempUserImage(tempUserImage)),
  setImageUpload: (imageUpload) => dispatch(setImageUpload(imageUpload)),
  setModalState: (modalState) => dispatch(setModalState(modalState))
})

export default connect (mapStateToProps, mapDispatchToProps)(UserProfileScreen)