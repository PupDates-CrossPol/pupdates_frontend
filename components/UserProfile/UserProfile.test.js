import React from 'react'
import renderer from 'react-test-renderer'
import * as actions from '../../actions/index'

import { UserProfileScreen as UserProfile, mapStateToProps, mapDispatchToProps } from './UserProfile'
import { ImageUpload } from '../ImageUpload/ImageUpload'
// import { mapStateToProps } from './UserProfile'

describe('<UserProfile />', () => {
  it.skip('has one child', () => {
    const mockUser = {
      first_name: 'sara',
      last_name: 'karsh',
      email: 'emailaddress',
      description: 'a string of words'
    }
    const tree = renderer.create(<UserProfile user={mockUser} />).toJSON()
    expect(tree.children.length).toBe(1)
  })

  it.skip('renders correctly', () => {
    const mockUser = {
      first_name: 'sara',
      last_name: 'karsh',
      email: 'emailaddress',
      description: 'a string of words'
    }
    const tree = renderer.create(<UserProfile user={mockUser} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})

describe('mapStateToProps', () => {
  it('should return an object with the user object', () => {
    const mockState = {
      user: {
        id: 1,
        first_name: 'sara',
        last_name: 'karsh',
        email: 'email',
        description: 'a string of words'
      },
      pack: [
        {
          id: 1,
          user_id: 1,
          name: 'fluffy',
        }
      ],
      packPhotos: [
        {
          id: 1,
          dog_id: 1,
          image_url: 'someaddress'
        }
      ],
      imageUpload: undefined,
      tempUserImage: undefined
    }

    const expectedState = {
      user: {
        id: 1,
        first_name: 'sara',
        last_name: 'karsh',
        email: 'email',
        description: 'a string of words'
      },
      pack: [
        {
          id: 1,
          user_id: 1,
          name: 'fluffy',
        }
      ],
      packPhotos: [
        {
          id: 1,
          dog_id: 1,
          image_url: 'someaddress'
        }
      ],
      imageUpload: undefined,
      tempUserImage: undefined
    }

    const mappedProps = mapStateToProps(mockState)

    expect(mappedProps).toEqual(expectedState)
  })
})

describe('mapDispatchToProps', () => {
  it('setUserInfo should be dispatched with the correct action', () => {
    const mockDispatch = jest.fn()

    const mockUser = {
      id: 2,
      first_name: 'bob',
      last_name: 'ross',
      email: 'bobrossemail',
      description: 'happy little trees'
    }

    const actionToDispatch = actions.setUserInfo(mockUser)
    const mappedDispatch = mapDispatchToProps(mockDispatch)
    mappedDispatch.setUserInfo(mockUser)
    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
  })
  it('setPackInfo should be dispatched with the correct action', () => {
    const mockDispatch = jest.fn()
    const mockPack = [
      {
        id: 1,
        user_id: 1,
        name: 'charles',
        breed: 'lab',
        size: 'large'
      }
    ]
    const actionToDispatch = actions.setPackInfo(mockPack)
    const mappedDispatch = mapDispatchToProps(mockDispatch)
    mappedDispatch.setPackInfo(mockPack)
    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
  })
  it('setPackPhotos should be dispatched with the correct action', () => {
    const mockDispatch = jest.fn()
    const mockPackPhotos = [
      {
        id: 1,
        dog_id: 1,
        image_url: 'someaddress'
      }
    ]
    const actionToDispatch = actions.setPackPhotos(mockPackPhotos)
    const mappedDispatch = mapDispatchToProps(mockDispatch)
    mappedDispatch.setPackPhotos(mockPackPhotos)
    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
  })
  it('setTempUserImage should be dispatched with the correct action', () => {
    const mockDispatch = jest.fn()
    const mockTempImage = {
      uri: 'someurl'
    }
    const actionToDispatch = actions.setTempUserImage(mockTempImage)
    const mappedDispatch = mapDispatchToProps(mockDispatch)
    mappedDispatch.setTempUserImage(mockTempImage)
    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
  })
})