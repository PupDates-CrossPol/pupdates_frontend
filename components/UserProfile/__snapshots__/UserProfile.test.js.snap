// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserProfile /> renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
>
  <View>
    <View>
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "flexDirection": "row",
            "marginLeft": 5,
            "opacity": 1,
            "padding": 10,
            "position": "absolute",
            "width": "25%",
          }
        }
      >
        <Text />
        <Text
          style={
            Object {
              "color": "rgb(53, 129, 252)",
              "fontSize": 25,
              "marginLeft": 5,
            }
          }
        >
          Menu
        </Text>
      </View>
      <Text
        style={
          Object {
            "alignSelf": "center",
            "color": "rgba(0,0,0,0.57)",
            "fontSize": 30,
            "fontWeight": "300",
            "padding": 10,
          }
        }
      >
        sara
      </Text>
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "borderBottomColor": "rgba(0,0,0,0.57)",
            "borderBottomWidth": 1,
            "elevation": 5,
            "height": 265,
            "justifyContent": "center",
            "marginLeft": 55,
            "marginRight": 55,
            "opacity": 1,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.25,
            "shadowRadius": 3.84,
          }
        }
      >
        <Image
          source={
            Object {
              "uri": undefined,
            }
          }
          style={
            Object {
              "aspectRatio": 1,
              "borderColor": "black",
              "borderRadius": 100,
              "borderWidth": 2,
              "height": 199,
            }
          }
        />
      </View>
      <Text
        style={
          Object {
            "color": "rgb(53, 129, 252)",
            "fontSize": 25,
            "marginBottom": 20,
            "marginLeft": 5,
            "marginTop": 20,
            "paddingLeft": 10,
          }
        }
      >
        About Me:
      </Text>
      <Text
        style={
          Object {
            "fontSize": 20,
            "fontWeight": "300",
            "marginTop": 5,
            "paddingLeft": 45,
          }
        }
      >
        First Name: 
        sara
      </Text>
      <Text
        style={
          Object {
            "fontSize": 20,
            "fontWeight": "300",
            "marginTop": 5,
            "paddingLeft": 45,
          }
        }
      >
        Last Name: 
        karsh
      </Text>
      <Text
        style={
          Object {
            "fontSize": 20,
            "fontWeight": "300",
            "marginTop": 5,
            "paddingLeft": 45,
          }
        }
      >
        Email: 
        emailaddress
      </Text>
      <Text
        style={
          Object {
            "color": "rgb(53, 129, 252)",
            "fontSize": 25,
            "marginBottom": 20,
            "marginLeft": 5,
            "marginTop": 20,
            "paddingLeft": 10,
          }
        }
      >
        Description:
      </Text>
      <Text
        style={
          Object {
            "fontSize": 20,
            "fontWeight": "300",
            "marginTop": 5,
            "paddingLeft": 45,
          }
        }
      >
        a string of words
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;
